### PR TITLE
fix favorites logic

### PR DESCRIPTION
**Pull request by DmitriyUshkvarok**

### CODE REVIEW

File: src/components/Product/ProductCard/ProductCard.tsx [hunk 1]
```diff
@@ -61,7 +57,8 @@ export default function ProductCard({
       </Link>
       <div className={' absolute right-0 top-0'}>
         <ButtonHeart
-          active={isActive}
+          // active={isActive}
```
Comment (annstriganova) [inline src/components/Product/ProductCard/ProductCard.tsx:60]: nit: do we need this commented out code?

File: src/components/Product/ProductCard/ProductCard.tsx [hunk 2]
```diff
@@ -19,19 +20,14 @@ export default function ProductCard({
...
-    try {
-      if (isActive) {
-        await removeFavourite(id, token)
-      } else {
-        await addFavourite(id, token)
-      }
-    } catch (error) {
-      console.error('Error in handleButtonClick:', error)
-    }
+    await handleFavouriteButtonClick(id, token, isInFavourites, isActive, addFavourite, removeFavourite)
```
Comment (annstriganova) [inline src/components/Product/ProductCard/ProductCard.tsx:30]: nit: this code seems to be repeated; not sure if it's possible to refactor, but I'd consider this in another PR

**Review by annstriganova — APPROVED** [review]
Approved.

File: src/components/Product/ProductCard/ProductCard.tsx (hunk 2 above)
Comment (DmitriyUshkvarok) [reply src/components/Product/ProductCard/ProductCard.tsx:30]: Absolutely right! This function with check is currently used in 4 different components in our project. That's why I put it in a separate utility and just call it in the necessary components by passing the necessary properties to it. Is it better to do with one line of code in a component or to write this huge function repeatedly in each component? Or what refactor do you mean?

handleButtonClick
handleFav

File: src/components/Product/ProductCard/ProductCard.tsx (hunk 1 above)
Comment (DmitriyUshkvarok) [reply src/components/Product/ProductCard/ProductCard.tsx:60]: Thank you! Deleted.

**Review by iakivpekarskyi — APPROVED** [review]
B

**Pull request merged**